### DISCUSSION TITLE
Gas turbine

### DISCUSSION THREAD

**Discussion by doimattertome** [oemof/tespy#511] [Q&A]
First of all, thank you very much for this program package and your hard work!
I'm having problems with a gas turbine computational simulation，I tried not specifying the heat input to the combustion chamber, but instead determining the heat input by fuel composition and mass flow rate, but it seems that the values are a bit on the large side, resulting in an error report.
as follows:
Invalid value for ti: ti = 4139810522892.0557 above maximum value (1000000000000.0) at component combustion chamber.
The solver does not seem to make any progress, aborting calculation. Residual value is 2.43e-01. This frequently happens, if the solver pushes the fluid properties out of their feasible range.

```python
from tespy.components import Sink, Source, CombustionChamber
from tespy.connections import Connection
from tespy.networks import Network
from tespy.tools.fluid_properties import T_sat_p
import shutil
nw = Network(p_unit='bar', T_unit='C', iterinfo=False)
amb = Source('ambient air')
sf = Source('fuel')
fg = Sink('flue gas outlet')
comb = CombustionChamber('combustion chamber')
comb.component()
'combustion chamber'
amb_comb = Connection(amb, 'out1', comb, 'in1')
sf_comb = Connection(sf, 'out1', comb, 'in2')
comb_fg = Connection(comb, 'out1', fg, 'in1')
nw.add_conns(sf_comb, amb_comb, comb_fg)
sf_comb.set_attr(m=84100)
comb.set_attr(lamb=1.5)
amb_comb.set_attr(p=1, T=20, fluid={'Ar': 0.0129, 'N2': 0.7553,
'CO2': 0.0004, 'O2': 0.2314})
sf_comb.set_attr(T=25, fluid={'CO2': 0.03, 'H2': 0.01, 'CH4': 0.96})
nw.solve('design')
comb_fg.set_attr(T=625)
comb.set_attr(lamb=None)
nw.solve('design')
nw.print_results()
```

**Comment by fwitte** [comment]
What would you expect by 84000 kg/s of methane?

**Comment by doimattertome** [comment]
Thank you! Sorry I got the units wrong, I thought the default was kg/h, but how do I check the units?

**Reply by fwitte** [reply]
It is not that obvious. All units are SI by default, the `Network` class will give you an error if you try to set unknown units and the list of available ones, kg / h should be available I think. Maybe a help for that would be nice then.

**Comment by doimattertome** [comment]
Hello, I would like to ask you which component I should use to simulate the direct injection of atomized cooling water into the tempered water vapor to evaporate it into water vapor and mix it well with the tempered water vapor to reduce the temperature of the tempered water vapor. thanks a lot!

**Reply by fwitte** [reply]
Hi @doimattertome,

you want to mix water vapor with liquid water? Then you can just use the `Merge` component... Does that work for you?

```python
from tespy.networks import Network
from tespy.connections import Connection
from tespy.components import Source, Sink, Merge


nwk = Network(p_unit="bar", T_unit="C")

vapor = Source("water vapor")
liquid = Source("injection water")
cooled = Sink("cooled water vapor")

merge = Merge("injection")

c1 = Connection(vapor, "out1", merge, "in1", label="1")
c2 = Connection(liquid, "out1", merge, "in2", label="2")
c3 = Connection(merge, "out1", cooled, "in1", label="3")

nwk.add_conns(c1, c2, c3)

c1.set_attr(fluid={"water": 1}, m=10, T=600, p=100)
c2.set_attr(fluid={"water": 1}, m=1, T=100)

nwk.solve("design")
```

**Reply by fwitte** [reply]
Did this answer your question?
